File: perl/conintl.pl
#!/usr/bin/perl
# perl/conintl.pl
# Ejecutar: plackup conintl.pl -p 5003
# URL: http://localhost:5003/?n=10
# Convierte un número a letras en español usando Lingua::ES::Numeros

use strict;
use warnings;
use Plack::Request;
use Lingua::ES::Numeros;

my $app = sub {
    my $env = shift;
    my $req = Plack::Request->new($env);
    
    my $n = $req->param('n') // 0;
    
    my $conv = Lingua::ES::Numeros->new(GENERO => 'M');
    my $texto = $conv->cardinal($n);
    
    return [
        200,
        ['Content-Type' => 'text/plain; charset=UTF-8'],
        [ $texto ]
    ];
};

# IMPORTANTE: Devolver la referencia de código
$app;

File: nodejs/conintl.js
// nodejs/conintl.js
// Uso: node conintl.js  -> http://localhost:3003/?n=10
// Convierte un número a letras en español usando written-number

const http = require('http');
const url = require('url');
const writtenNumber = require('written-number');

const server = http.createServer((req, res) => {
  const { query } = url.parse(req.url, true);
  const n = parseInt(query.n) || 0;

  try {
    // Convertir número a letras en español
    const texto = writtenNumber(n, { lang: 'es' });

    res.writeHead(200, { 'Content-Type': 'text/plain; charset=utf-8' });
    res.end(texto);
  } catch (err) {
    console.error('Error:', err);
    res.writeHead(500);
    res.end(`Error: ${err.message}`);
  }
});

server.listen(3003, () => console.log('http://localhost:3003/?n=10'));